### ISSUE TITLE
Be able to use commando to insert classes into files.

### ISSUE THREAD

**Issue by edster** [Komodo/KomodoEdit#887]
I would like to be able to use Commando to include classes such as `use Anomaly\FilesModule\Disk\DiskModel;`

Instead of manually having to write out the class and path, it would be nice to be able to type out the file `DiskModel` then have a key I can press that will then generate the `use` and insert it.
Eg,
```
use Anomaly\Streams\Platform\Addon\Plugin\Plugin;
use Anomaly\FilesModule\Disk\DiskModel;

class RandomFilePlugin extends Plugin
{
}
```

**Comment by edster** [comment]
In the mean time, how easy would this be for me to customize myself?

**Comment by Naatan** [comment]
I don't see what Commando would do with your given use-case, this would just be a userscript that could run on a shortcut.

To the point though; you say you want it to automatically resolve the full namespace from just the last bit. This would take codeintel to try and figure out what possible namespace prefix are available for the given suffix. To my knowledge this would require codeintel to be able to resolve things in a backwards manner, which it currently does not. @mitchell-as can you comment?

**Comment by Naatan** [comment]
For the mean time, the auto resolving part is currently not possible. However if you can make certain assumptions with your PHP framework (you probably can) you could write a little PHP script that resolves a namespace suffix to the full namespace path and invoke that script from a komodo userscript.

Happy to explain this further, but please create a thread on the forums for this, as it's not what we use our bug tracker for.

**Comment by mitchell-as** [comment]
In theory we might be able to lookup a class name in the codeintel db and try inserting based on that, but if there is some sort of autoloading mechanism that determines namespaces based on a particular convention (not in the code itself), then there is no way to do this at the moment. Does that help @Naatan?

**Comment by edster** [comment]
I have this in my codebase if it is relevant

https://github.com/barryvdh/laravel-ide-helper

**Comment by Naatan** [comment]
@mitchell-as yes, and to be clear I don't think this could work as a perfect 1:1 translation of suffix to namespace, rather the suffix could serve as a seed for the namespace search and if only one result is found then we could assume that's the right one. If multiple results are found then we'd have to give some type of prompt (autocompletion popup?).

It was my impression though that we would need to make some database improvements should we want to look for eg. classes in the project context (ie. not just the current file)?

**Comment by mitchell-as** [comment]
I agree there needs to be a way to resolve a class name outside the current file. It's possible codeintel has this capability already. We'd just have to narrow down multiple results like you suggested.

**Comment by mitchell-as** [comment]
Note to self: this might be done by creating a new trigger that asks for a lookup for the given class/namespace/interface name. The evaluator should search for the given symbol and return a list of "use" completions. Komodo would then present this list to the user for insertion.

**Comment by mitchell-as** [comment]
I've spent some time investigating this, and our current database structure does not allow us to to a blanket search of classes (much less determine their namespace nesting) in order to have Commando supply such a list for easy insertion. Extending the database to insert/query this information doesn't appear to be trivial and I'm not really sure where I'd begin on it. If we ever get around to overhauling our database, we can revisit this ticket, but for now I'm pushing into the backlog.
